action.yaml

```diff
@@ -53,55 +53,129 @@ runs:
-                --location '${{ steps.resolve-deploymentconfig.outputs.location }}' \
-                --verbose \
-                ${{ (inputs.what-if == 'true') && '--what-if' || '' }} \
-                ${{ (runner.debug == '1') && '--debug' || '' }} 2>stderr.txt | tee deployment.md)
-          fi
-
-          sed -i -e '1s/^/```diff\n/' -e '$a```' -e 's/~/!/g' -e 's/^[ \t]*//' deployment.md
-
-          if grep -q "ERROR:" stderr.txt; then
-              echo "::error::An error occurred during deployment. See error details in below"
-              cat stderr.txt
-              exit 1
-          fi
 
+          #* Multiline annotation replacement function
+          format_multiline_annotation() {
+            # Input: A multiline string
+            # Output: The string formatted for GitHub annotations
+            
+            local input="$1"
+            
+            # Replace newlines with %0A to support multiline annotations in GitHub workflows
+            local formatted_string=$(echo "$input" | sed ':a;N;$!ba;s/\n/%0A/g')
+            
+            # Output the formatted string
+            echo "$formatted_string"
+          }
+
+          #* Cleanup function
+          print_stderr() {
+            #* Read the file line by line
+            while IFS= read -r line; do
+              if [[ $line == WARNING:* ]]; then
+                #* Extract the WARNING message and print it in yellow
+                echo -e "\e[33m${line#WARNING: }\e[0m"
+              elif [[ $line == INFO:* ]]; then
+                #* Print the INFO message in the default color
+                echo "${line#INFO: }"
+              elif [[ $line == ERROR:* ]]; then
+                #* Extract the JSON from the ERROR message
+                json=${line#ERROR: }
+                #* Pretty-print the JSON and apply red color to each line, suppressing extra lines
+                formatted_json=$(echo "$json" | jq . 2>/dev/null | while IFS= read -r json_line || [[ -n $json_line ]]; do
+                  echo -e "\e[31m${json_line}\e[0m"
+                done)
+                annotation_json=$(format_multiline_annotation "$formatted_json")
+                echo "::error::An error occurred during deployment. See error details in below%0A${annotation_json}"
+              fi
+            done < stderr.log
+          }
+
+          #* Set trap to call the cleanup function on ERR
+          trap print_stderr ERR
+
+          #* Run deployment command
+          deploymentOutput=$({
+            case $scope in
+              "resourceGroup")
+                az deployment group create \
+                  --parameters "$parameter_file" \
+                  --name "$deployment_name" \
+                  --resource-group "$resource_group" \
+                  --verbose \
+                  $( [ $what_if = 'true' ] && echo "--what-if" ) \
+                  $( [ $debug = '1' ] && echo "--debug" )
```

Comment:
(nitpicking warning): integer instead of string comparison can be done with [[ $debug -eq 1 ]]
also nice to be consistent with double bracket bash tests and double equals symbols (if left as a string comparison) for consistency sake with code above